packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectSearchService.java

```diff
@@ -0,0 +1,205 @@
+package software.uncharted.terarium.hmiserver.service.data;
+
+import co.elastic.clients.elasticsearch._types.FieldSort;
+import co.elastic.clients.elasticsearch._types.SortOptions;
+import co.elastic.clients.elasticsearch._types.SortOrder;
+import co.elastic.clients.elasticsearch._types.query_dsl.HasChildQuery;
+import co.elastic.clients.elasticsearch._types.query_dsl.Query;
+import co.elastic.clients.elasticsearch._types.query_dsl.QueryBuilders;
+import co.elastic.clients.elasticsearch.core.SearchRequest;
+import java.io.IOException;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.UUID;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import software.uncharted.terarium.hmiserver.configuration.ElasticsearchConfiguration;
+import software.uncharted.terarium.hmiserver.models.dataservice.project.Project;
+import software.uncharted.terarium.hmiserver.service.elasticsearch.ElasticsearchService;
+
+@RequiredArgsConstructor
+@Service
+@Slf4j
+public class ProjectSearchService {
+
+	protected final ElasticsearchConfiguration elasticConfig;
+	protected final ElasticsearchService elasticService;
+
+	protected String getIndex() {
+		log.info("Project index: {}", elasticConfig.getProjectIndex());
+		return elasticConfig.getProjectIndex();
+	}
+
+	public String getAlias() {
+		log.info("Project alias: {}", elasticConfig.getProjectAlias());
+		return elasticConfig.getProjectAlias();
+	}
+
+	public String getCurrentAssetIndex() throws IOException {
+		return elasticService.getIndexFromAlias(getAlias());
+	}
+
+	@Data
+	public static class PermissionJoin {
+
+		private String name = null;
+		private UUID parent = null;
+
+		static PermissionJoin CreateProjectJoin() {
+			final PermissionJoin join = new PermissionJoin();
+			join.setName("project");
+			join.setParent(null);
+			return join;
+		}
+
+		static PermissionJoin CreatePermissionJoin(final UUID projectId) {
```

Comment:
should this not be `createPermissionJoin()` as it is not the constructor for PermissionJoin